internal/services/tesla_fleet_api_service.go

```diff
@@ -91,50 +90,100 @@ func (t *teslaFleetAPIService) CompleteTeslaAuthCodeExchange(ctx context.Context
 	}, nil
 }
 
-// GetVehicles godoc
+// GetVehicles
 // @Description Call Tesla Fleet API to get a list of vehicles using authorization token
-// @Param       token - authorization token to be used as bearer token
-// @Param       region - API region which is used to determine which fleet api to call
-// @Returns     {array} []services.TeslaVehicle
 func (t *teslaFleetAPIService) GetVehicles(ctx context.Context, token, region string) ([]TeslaVehicle, error) {
 	baseURL := fmt.Sprintf(t.Settings.Tesla.FleetAPI, region)
 	url := baseURL + "/api/1/vehicles"
 
+	resp, err := t.performTeslaGetRequest(ctx, url, token)
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch vehicles for user: %w", err)
+	}
+	defer resp.Body.Close()
+
+	vehicles := new(GetVehiclesResponse)
+	if err := json.NewDecoder(resp.Body).Decode(vehicles); err != nil {
+		return nil, fmt.Errorf("invalid response encountered while fetching user vehicles: %w", err)
+	}
+
+	if vehicles.Response == nil {
+		return nil, fmt.Errorf("error occurred fetching user vehicles")
+	}
+
+	return vehicles.Response, nil
+}
+
+// GetVehicle
+// @Description Call Tesla Fleet API to get a single vehicle by ID
+func (t *teslaFleetAPIService) GetVehicle(ctx context.Context, token, region string, vehicleID int) (*TeslaVehicle, error) {
+	baseURL := fmt.Sprintf(t.Settings.Tesla.FleetAPI, region)
+	url := fmt.Sprintf("%s/api/1/vehicles/%d", baseURL, vehicleID)
```

Comment:
nit: [path](https://pkg.go.dev/path) is preferred when creating a URLs from varibles.
```suggestion
	url := path.Join(baseURL, "api/1/vehicles", vehicleID)
```

Reply:
Any advantage to

https://pkg.go.dev/net/url#JoinPath

?

Reply:
> nit: [path](https://pkg.go.dev/path) is preferred when creating a URLs from varibles.

I guess you would have to convert `vehicleID` to a string, first.

Reply:
Since this is a nit, I'd rather stick to `fmt.Sprintf`, matter of preference.